examples/hubert/lightning.py

```diff
@@ -0,0 +1,127 @@
+from typing import Tuple
+
+import torch
+import torchaudio
+from dataset import BucketizeSampler, DistributedBatchSampler, HuBERTDataSet, CollateFnHubert
+from loss import hubert_loss
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.optim.optimizer import Optimizer
+from torch.utils.data import DataLoader
+
+
+Batch = Tuple[Tensor, Tensor, Tensor]
+
+
+class PolynomialDecayLRScheduler(torch.optim.lr_scheduler._LRScheduler):
```

Comment:
is the following correct?
- for training iterations in [0, `warmup_updates`], we maintain a constant learning rate of `base_lr / warmup_updates`
- in (`warmup_updates`, `max_updates`), we linearly decrease learning rate from ~`base_lr` to 0
- `max_updates` and beyond, we hold the learning rate at 0

if so, what does "polynomial" refer to? and is it expected that the learning rate jumps from `base_lr / warmup_updates` to roughly `base_lr` after the `warmup_updates`-th iteration?

Reply:
This LRScheduler is from [fairseq](https://github.com/pytorch/fairseq/blob/fcca32258c8e8bcc9f9890bf4714fa2f96b6b3e1/fairseq/optim/lr_scheduler/polynomial_decay_schedule.py#L74-L89). The logic should be like this:
- from [0, warmup_updates] steps, increase the learning rate from initial lr to lr * num_step / warmup_updates
- from [warmup_updates, max_updates], decrease the learning rate by the formula.
- from [max_updaes, ], hold the learning rate at 0.

Reply:
hmm, the impl doesn't seem correct. num_step / warm_updates is smaller than 1, how can they call it warmup then...

Reply:
also, perhaps we should call this something other than "polynomial" since it isn't applying a polynomial function to `pct_remaining` as in the original implementation